pkg/vgpu/config.go

```diff
@@ -45,6 +56,152 @@ func NewNvlibVGPUConfigManager() Manager {
 	return &nvlibVGPUConfigManager{nvlib.New()}
 }
 
+func (m *nvlibVGPUConfigManager) GetVGPUConfigforVFIO(gpu int) (types.VGPUConfig, error) {
+	nvdevice, err := m.nvlib.Nvpci.GetGPUByIndex(gpu)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get GPU by index %d: %v", gpu, err)
+	}
+	vgpuConfig := types.VGPUConfig{}
+	vfnum := 0
+	if nvdevice.SriovInfo.PhysicalFunction == nil {
+		return vgpuConfig, nil
+	}
+	totalVF := int(nvdevice.SriovInfo.PhysicalFunction.TotalVFs)
+	for vfnum < totalVF {
+		vfAddr := HostPCIDevicesRoot + "/" + nvdevice.Address + "/virtfn" + strconv.Itoa(vfnum) + "/nvidia"
+		if _, err := os.Stat(vfAddr); err != nil {
+			vfnum++
+			continue
+		}
+		vgpuTypeNumberBytes, err := os.ReadFile(vfAddr + "/current_vgpu_type")
+		if err != nil {
+			return nil, fmt.Errorf("unable to read current vGPU type: %v", err)
+		}
+		vgpuTypeNumber, err := strconv.Atoi(strings.TrimSpace(string(vgpuTypeNumberBytes)))
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert current vGPU type to int: %v", err)
+		}
+		if vgpuTypeNumber == 0 {
+			vfnum++
+			continue
+		}
+		vgpuTypeName, err := m.getVGPUTypeNameforVFIO(vfAddr + "/creatable_vgpu_types", vgpuTypeNumber)
```

Comment:
Since we are only reaching this point when `vgpuTypeNumber != 0`, won't this file always be empty?